### PR TITLE
feat(ui): Inject Custom CSS

### LINKED ISSUE #1451: UI: Add ability to visually distinguish between deployment environments
**Is this a BUG REPORT or FEATURE REQUEST?**: **FEATURE REQUEST**

We run a dev environment and a production environment, which is fairly typical. We currently access the `argo-ui` through a proxy cmd, and I think it's easy enough to not specific the environment (dev/prod) properly and get confused as both look visually the same. 

What do you think about this problem? If/When we place domains in front of the UIs,  eg. `dev-argo-ui.company-ds.com` and `prod-argo-ui.company-ds.com` I think this issue will be alleviated, but not totally fixed as no element exists in the UI to display the deployment environment.

### PR DESCRIPTION
**Pull request by dtaniwaki**
Checklist:

* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo-workflows/blob/master/USERS.md).

Screen Shot 2021-07-05 at 12 41 58 PM

Fixes #1451

You can test this feature by executing the following commands.

```bash
$ kubectl -n argo create configmap custom-css --from-literal "custom.css=.nav-bar { background-color: #2196f3; }"
$ kubectl -n argo patch configmap workflow-controller-configmap -p '{"data":{"ui.cssurl":"./extra/custom.css"}}'
$ kubectl -n argo patch deployment argo-server -p '{"spec":{"template":{"spec":{"containers":[{"name":"argo-server","args":["server", "--secure=false", "--auth-mode=hybrid", "--extra-files-dir=/shared/app/custom"], "volumeMounts": [{"name": "custom-css", "mountPath": "/shared/app/custom"}]}], "volumes": [{"name":"custom-css", "configMap": {"name": "custom-css"}}]}}}}'

```

### CODE REVIEW

File: server/static/static.go [hunk 1]
```diff
@@ -6,29 +6,27 @@ import (
 	"strings"
 )
 
+var (
+	extraPath = "/extra/"
```
Comment (dtaniwaki) [inline server/static/static.go:10]: Any good path for extra files?

**Comment by dtaniwaki** [comment]
I updated the code so the server doesn't return 304 on removed extra files.

**Comment by alexec** [comment]
I suggest doing this different.

1. Add a new variable to get info called “env”. This can be “dev” , “e2e” or “prod”.
2.  Use this to load a CSS file based on the name. Eg dev.css
3. Bundle CSS using those conventional names.
4. Add a new parameter to Argo server, “—env” or “ARGO_ENV”

**Comment by dtaniwaki** [comment]
Thank you for your suggestion. I have 2 questions for the env idea. 

What env should we use for a locally running argo server?

We could have 2 argo servers with different instance ID in the production. Should we use different env names like prod_1 and prod_2?

**Comment by dtaniwaki** [comment]
By the way, as for the env-like name, I injected it by the following CSS.

```
    .nav-bar {
      background-color: #3f51b5;
    }
    .nav-bar__logo::before {
      content: 'stg';
      color: white;
      text-align: center;
      display: block;
    }
```

And got the following result.

Screen Shot 2021-07-06 at 10 55 04 PM

**Comment by dtaniwaki** [comment]
However, the env idea is also good. I'd like to hear other people's thoughts.

**Review by alexec — CHANGES_REQUESTED** [review]
See comments.

**Comment by dtaniwaki** [comment]
Hmm... No other comments.

@jessesuen Do you have any thought about the compatibility with ArgoCD?

**Comment by kenchan0130** [comment]
Is there any progress on this?
I'd like to apply CSS with `values` as well as Argo CD.

**Comment by tooptoop4** [comment]
where @dtaniwaki

**Pull request closed without merging**